tfe/resource_tfe_policy_set_test.go

```diff
@@ -392,6 +394,154 @@ func TestAccTFEPolicySet_updateVCSBranch(t *testing.T) {
 	})
 }
 
+func TestAccTFEPolicySet_version(t *testing.T) {
+	skipIfFreeOnly(t)
+
+	policySet := &tfe.PolicySet{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	checksum, err := hashPolicies(testFixtureVersionFiles)
+	if err != nil {
+		t.Fatalf("Unable to generate checksum for policies %v", err)
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEPolicySetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEPolicySet_versionbasic(rInt, testFixtureVersionFiles),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEPolicySetExists("tfe_policy_set.foobar", policySet),
+					testAccCheckTFEPolicySetAttributes(policySet),
+					resource.TestCheckResourceAttr(
+						"tfe_policy_set.foobar", "name", "tst-terraform"),
+					resource.TestCheckResourceAttr(
+						"tfe_policy_set.foobar", "description", "Policy Set"),
+					resource.TestCheckResourceAttr(
+						"tfe_policy_set.foobar", "slug.%", "3"),
+					resource.TestCheckResourceAttrSet(
+						"tfe_policy_set.foobar", "slug.source_path"),
+					resource.TestCheckResourceAttr(
+						"tfe_policy_set.foobar", "slug.source_path", testFixtureVersionFiles),
+					resource.TestCheckResourceAttrSet(
+						"tfe_policy_set.foobar", "slug.checksum"),
+					resource.TestCheckResourceAttr(
+						"tfe_policy_set.foobar", "slug.checksum", checksum),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTFEPolicySet_version_updateCreate(t *testing.T) {
+	skipIfFreeOnly(t)
+
+	policySet := &tfe.PolicySet{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	originalChecksum, err := hashPolicies(testFixtureVersionFiles)
+	if err != nil {
+		t.Fatalf("Unable to generate checksum for policies %v", err)
+	}
+
+	newFile := fmt.Sprintf("%s/newfile.test.sentinel", testFixtureVersionFiles)
+	removeFile := func() {
+		// This func is used below, that is why it is not an anonymous function.
+		// It is used because if there is a test fatal (t.Fatal), then defer does
+		// not get called. So we call this `removeFile` function both in the defer
+		// and explicitly below.
+		err := os.Remove(newFile)
+		if err != nil {
+			t.Fatalf("Error removing file %v", err)
+		}
+	}
```

Comment:
Not sure I follow. Do you mean that having the extra `removeFile` function call below as a precaution isn't that useful because this would happen to any test fatal?

Reply:
Basically, yeah. It's a fix for this particular test but is something inherent to any of them. Harmless to leave, just figured I'd say I wouldn't expect it as that's a bigger issue with the whole suite 😩